client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -176,7 +246,49 @@ export class BlockEditorEditingPresentational extends React.Component<
     }
   };
 
-  private onKeyDown = event => {
+  private onKeyDown = (event, change) => {
```

Comment:
Add types?

Reply:
Also, `onKeyDown` is a bit difficult to reason about because it's a long sequence of conditionals and actions (that are themselves kind of difficult to think about for someone like me who doesn't know Slate well). I wonder if there are some abstractions that could make the function easier to read, and that could make it more obvious that it's behaving correctly.

Reply:
I cleaned it up pretty significantly here: https://github.com/oughtinc/mosaic/pull/145/commits/d6197a53bfccb542974d2501f05914dc084db102. Still need to add types and think about best way to add abstraction.

Reply:
OK added types here https://github.com/oughtinc/mosaic/pull/145/commits/53d0c601af59cb7397814b4cc94ab4fcece3fd4d

After this branch is merged, I'll make a branch where I just clean up all the types. At least put them in a dedicated folder/file so we avoid duplicate work in the future.